### ISSUE TITLE
routes without id don't work

### ISSUE THREAD

**Issue by jensklose** [nilportugues/laravel5-hal-json-transformer#3]
lumen.ERROR: exception 'NilPortugues\Api\Mapping\MappingException' with message 'Could not find "id_properties" property with data .

but an api should have such routes ;)
I tried to test your work with the entry point of my API. "/api"

What is the undocumented way?

**Comment by nilportugues** [comment]
There's no need for the hal+json transformer there. `/api` should point to your API documentation the list of endpoints available. Using the response trait and a json_encode should be OK. 

If you intend to do `/api/user/1/friends` for instance, `1` should be a `userId`, that is constructor parameter for a getAction in your controller.

The hal+json transformer is to be used for mapping resources (eg: eloquent models). All models have an id, so that's when the hal-json transformer should be used.

The transformer  is not a API server. That's just another project I will eventually release. 

In conclusion, for URLs containing ids, use the hal-json transformer, elsewhere just normal arrays that can be json_encoded+response trait. You should be fine.

**Comment by jensklose** [comment]
I think you have a gap in your approach.
If I design an API then I provide an interface. IMO this interface should be consistent.

What about all listing resources without an ID?
/api/oders
/api/customers

```
    {
        "_links": {
            "self": { "href": "/api/orders" },
            "next": { "href": "/api/orders?page=2" }
        },
        "_embedded": {
            "ea:order": [{
                 /* ...*/
    }
```

All of this should be possible to respond a HAL JSON and the whole API should be viewable in HAL-Browser, I guess.

Now I understand your projects as a single resource transformer. Please update the documentation with a hint.

**Comment by nilportugues** [comment]
It's actually feasible if I just remove the check for ids in the source code.

**Comment by nilportugues** [comment]
@jensklose should work now. Just do a `composer update` and let me know if I can help you.

**Comment by jensklose** [comment]
Hey, I'll do so ~~if you share it.
There are no changes in the repository. ;)~~

**Comment by nilportugues** [comment]
@jensklose cool :+1:

**Comment by jensklose** [comment]
first feedback:
It seems to work partially;
debugging shows the rendered content
but
nothing was sent to client

```
HTTP/1.1 200 OK
Server: nginx
Date: Fri, 02 Oct 2015 17:19:58 GMT
Content-Type: application/hal+json
Transfer-Encoding: chunked
Connection: keep-alive
X-Powered-By: PHP/5.5.7 ZendServer/6.3.0
...
X-Zend-Debug-Server: OK
Cache-Control: max-age=0, must-revalidate, private
Strict-Transport-Security: max-age=31536000; includeSubdomains
X-Content-Type-Options: nosniff
```

**Comment by nilportugues** [comment]
Make sure you have the latest packages by doing `php composer update`.

If still not working, mind sharing your mapping and classes in a gist?  Even without ids it will render information..I've investigated but haven't been able to reproduce your case.

**Comment by nilportugues** [comment]
I'm closing this as no gist has been provided. Will be reopened if gist if provided, or more data on the subject.

For those reading, input without ID field specified will still output data where possible and return a valid HAL+JSON response.

**Comment by jensklose** [comment]
Thanks for your investigation.
Some other issues with Lumen/Laravel forces a switch to Slim framework.

**Comment by nilportugues** [comment]
You can always use the standalone package... will certainly get the job done :)
